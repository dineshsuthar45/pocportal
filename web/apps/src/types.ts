import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
