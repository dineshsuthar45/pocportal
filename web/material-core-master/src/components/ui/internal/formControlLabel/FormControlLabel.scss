.m-form-control-label {
	display: flex;

	&.top {
		flex-direction: column;
	}

	&.end {
		flex-direction: row-reverse;
		justify-content: flex-end;
	}

	&.bottom {
		flex-direction: column-reverse;
	}

	&.start {
		flex-direction: row;
	}

	&.start,
	&.end {
		align-items: center;
	}

	&.disabled {
		cursor: default;
		color: rgba($color: transparent, $alpha: 0.3);

		&.dark {
			color: rgba($color: white, $alpha: 0.3);
		}

		.tooltip-label {
			cursor: default;
		}
	}

	.tooltip-label {
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;

		cursor: pointer;
	}
}
