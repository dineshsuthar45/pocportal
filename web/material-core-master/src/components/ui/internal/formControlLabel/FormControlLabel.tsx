import { useTheme } from "@mui/material";
import classNames from "classnames";
import { useEffect, useState } from "react";
import { MTooltip } from "../../tooltip";
import "./FormControlLabel.scss";
import { FormControlLabelProps } from "./FormControlLabel.types";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/form-label/
 */
export const MFormControlLabel = ({
  children,
  tooltipProps,
  label = "",
  labelPlacement = "top",
  disabled = false,
}: FormControlLabelProps) => {
  const theme = useTheme();
  const [tooltipPlacement, setTooltipPlacement] = useState(
    tooltipProps.placement ?? "top"
  );

  useEffect(() => {
    setTooltipPlacement(tooltipProps.placement ?? "top");
  }, [tooltipProps.placement]);

  return (
    <div
      className={classNames("m-form-control-label", labelPlacement, {
        disabled,
        dark: theme.palette.mode === "dark",
      })}
    >
      <MTooltip
        className={classNames("tooltip-label")}
        placement={tooltipPlacement}
        {...tooltipProps}
      >
        <label>{label}</label>
      </MTooltip>
      <div className="label-children">{children}</div>
    </div>
  );
};

MFormControlLabel.defaultProps = {
  label: "",
  labelPlacement: "top",
  disabled: false,
};
