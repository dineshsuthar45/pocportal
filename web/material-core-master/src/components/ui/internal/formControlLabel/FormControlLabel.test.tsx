import { render } from "@testing-library/react";
import React from "react";
import { MFormControlLabel } from "./FormControlLabel";
import { FormControlLabelProps } from "./FormControlLabel.types";

describe("MFormControlLabel", () => {
  const renderComponent = () => render(<MFormControlLabel {...props} />);
  let props: FormControlLabelProps;

  beforeEach(() => {
    props = {
      label: "Component label",
      tooltipProps: {
        title: "Text on hover",
        showAlways: true,
      },
      children: <div>This is a custom component</div>,
    };
  });

  it("should render correctly", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it("should render with no tooltip", () => {
    props.tooltipProps.showAlways = false;
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
