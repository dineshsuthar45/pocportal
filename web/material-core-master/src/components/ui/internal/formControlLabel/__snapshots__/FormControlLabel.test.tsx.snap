// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MFormControlLabel should render correctly 1`] = `
<div>
  <div
    class="m-form-control-label top"
  >
    <div
      aria-label="Text on hover"
      class="m-tooltip-root tooltip-label tooltip-container"
      data-mui-internal-clone-element="true"
    >
      <label>
        Component label
      </label>
    </div>
    <div
      class="label-children"
    >
      <div>
        This is a custom component
      </div>
    </div>
  </div>
</div>
`;

exports[`MFormControlLabel should render with no tooltip 1`] = `
<div>
  <div
    class="m-form-control-label top"
  >
    <div
      aria-label=""
      class="m-tooltip-root tooltip-label tooltip-container"
      data-mui-internal-clone-element="true"
    >
      <label>
        Component label
      </label>
    </div>
    <div
      class="label-children"
    >
      <div>
        This is a custom component
      </div>
    </div>
  </div>
</div>
`;
